bot/exts/help_channels/_cog.py

```diff
@@ -465,3 +484,32 @@ async def wait_for_dormant_channel(self) -> discord.TextChannel:
         self.queue_tasks.remove(task)
 
         return channel
+
+    async def update_available_help_channels(self) -> None:
+        """Updates the dynamic message within #how-to-get-help for available help channels."""
+        if not self.available_help_channels:
+            self.available_help_channels = set(
+                c for c in self.available_category.channels if not _channel.is_excluded_channel(c)
+            )
+
+        available_channels = AVAILABLE_HELP_CHANNELS.format(
+            available=', '.join(c.mention for c in self.available_help_channels) or None
+        )
+
+        if self.dynamic_message is not None:
+            try:
+                await self.bot.http.edit_message(
+                    constants.Channels.how_to_get_help, self.dynamic_message, content=available_channels
+                )
+                log.trace("Help channels have changed, dynamic message has been edited.")
```

Comment:
I think it's better for this log to come before the edit call. If it's after, nothing will be logged if the call fails.

Reply:
I actually have a different opinion on this one, I think the log should come after, since the whole purpose of logging is to record events, ie. if we see the log it means *x* happened, if it doesn't then we never reached that point.

Reply:
I'd log before the edit, then both in the `except` and the `else` clause to document the outcome. :smile:

Reply:
I'll put it above the edit log.